flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -221,6 +221,14 @@ class PFTBuilder {
   /// Initialize a new construct and make it the builder's focus.
   template <typename A>
   bool enterConstructOrDirective(const A &construct) {
+    // First handle top level directive case
+    if constexpr (std::is_same_v<parser::CompilerDirective, A>) {
+      if (parentVariantStack.size() == 1) {
```

Comment:
Instead of a depth check, it would be more robust to use `parentVariantStack.back().isA<lower::pft::Program>()`

Reply:
That's better thanks.